cpp-throughput/throughput_scaling.cpp

```diff
@@ -17,20 +17,28 @@ bool get_cluster_flag() {
 void run_throughput(std::ofstream& timing_file,
                     size_t n_bytes)
 {
+    std::string context("Run Throughput");
     int rank;
     MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+    log_data(context, LLDebug, "rank: " + std::to_string(rank) + " initiated");
 
     if(!rank)
+        log_data(context, LLInfo, "Connecting clients");
         std::cout<<"Connecting clients"<<std::endl<<std::flush;
 
     double constructor_start = MPI_Wtime();
     bool cluster = get_cluster_flag();
-    SmartRedis::Client client(cluster);
+    SmartRedis::Client client(cluster, context);
     double constructor_end = MPI_Wtime();
     double delta_t = constructor_end - constructor_start;
     timing_file << rank << "," << "client()" << ","
                 << delta_t << "\n";
 
+    //print rank # storing client() for debugging
+    std::string client_text = "client() time stored for rank: ";
+    client_text += std::to_string(rank);
+    log_data(context, LLDebug, client_text);
```

Comment:
With rank in context, this becomes `log_data(context, LLDebug, "client() time stored");`

Reply:
I see what u are saying now